mono/mini/driver.c

```diff
@@ -2754,7 +2754,9 @@ mono_main (int argc, char* argv[])
 MonoDomain * 
 mono_jit_init (const char *file)
 {
-	return mini_init (file, NULL);
+	MonoDomain *ret = mini_init (file, NULL);
+	MONO_ENTER_GC_SAFE_UNBALANCED;
```

Comment:
I think we need a comment here to explain why we want to leave this thread in GC Safe mode.